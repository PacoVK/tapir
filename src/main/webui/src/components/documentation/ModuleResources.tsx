import React from "react";
import { Box, Chip, Stack, Typography } from "@mui/material";
import { ModuleResourcesProps } from "../../types";

const ModuleResources = (props: ModuleResourcesProps) => {
  const { resources } = props;
  return (
    <>
      <Typography variant="h4">Resources</Typography>
      <Stack justifyContent="center" alignItems="left" spacing={2}>
        <Typography sx={{ mt: "1em" }} width={"50%"}>
          This is the list of resources that the module may create. The module
          can create zero or more of each of these resources depending on the
          count value. The count value is determined at runtime. The goal of
          this page is to present the types of resources that may be created.
        </Typography>
        <Typography width={"50%"}>
          This list contains all the resources this plus any submodules may
          create. When using this module, it may create fewer resources if you
          use a submodule.
        </Typography>
        <Typography>
          This module defines <strong>{resources.length}</strong> resources.
        </Typography>
        <ul>
          {resources.map((resource, index) => {
            return (
              <Box key={`module-resource-${index}`}>
                <li>
                  <Chip
                    sx={{ m: "0.5em" }}
                    label={`${resource.type}.${resource.name}`}
                    variant="outlined"
                  />
                </li>
              </Box>
            );
          })}
        </ul>
      </Stack>
    </>
  );
};

export default ModuleResources;
