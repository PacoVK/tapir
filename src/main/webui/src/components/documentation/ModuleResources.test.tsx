import { render } from "@testing-library/react";
import ModuleResources from "./ModuleResources";

describe("<ModuleResources /> spec", () => {
  it("renders the ModuleResources", () => {
    const view = render(
      <ModuleResources
        resources={[
          {
            name: "this",
            type: "ecr_lifecycle_policy",
          },
        ]}
      />,
    );
    expect(view).toMatchSnapshot();
  });
});
