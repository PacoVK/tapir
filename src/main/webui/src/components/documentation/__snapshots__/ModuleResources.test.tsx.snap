// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModuleResources /> spec renders the ModuleResources 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <h4
        class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
      >
        Resources
      </h4>
      <div
        class="MuiStack-root css-c08hs3-MuiStack-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body1 css-tialbm-MuiTypography-root"
        >
          This is the list of resources that the module may create. The module can create zero or more of each of these resources depending on the count value. The count value is determined at runtime. The goal of this page is to present the types of resources that may be created.
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-bbf04n-MuiTypography-root"
        >
          This list contains all the resources this plus any submodules may create. When using this module, it may create fewer resources if you use a submodule.
        </p>
        <p
          class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
        >
          This module defines 
          <strong>
            1
          </strong>
           resources.
        </p>
        <ul>
          <div
            class="MuiBox-root css-0"
          >
            <li>
              <div
                class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-outlinedDefault css-16m7pga-MuiChip-root"
              >
                <span
                  class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
                >
                  ecr_lifecycle_policy.this
                </span>
              </div>
            </li>
          </div>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <h4
      class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
    >
      Resources
    </h4>
    <div
      class="MuiStack-root css-c08hs3-MuiStack-root"
    >
      <p
        class="MuiTypography-root MuiTypography-body1 css-tialbm-MuiTypography-root"
      >
        This is the list of resources that the module may create. The module can create zero or more of each of these resources depending on the count value. The count value is determined at runtime. The goal of this page is to present the types of resources that may be created.
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-bbf04n-MuiTypography-root"
      >
        This list contains all the resources this plus any submodules may create. When using this module, it may create fewer resources if you use a submodule.
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 css-ahj2mt-MuiTypography-root"
      >
        This module defines 
        <strong>
          1
        </strong>
         resources.
      </p>
      <ul>
        <div
          class="MuiBox-root css-0"
        >
          <li>
            <div
              class="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-outlinedDefault css-16m7pga-MuiChip-root"
            >
              <span
                class="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
              >
                ecr_lifecycle_policy.this
              </span>
            </div>
          </li>
        </div>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
